File: MenuVoting.WebApi/MenuVotingApp/src/app/models/menu-pool/menu-pool.ts
import { Menu } from "../menu/menu";

export class MenuPool {
  id: string | null;
  restaurantId: string | null;
  menus: Menu[];
  constructor(id: string | null, menus: Menu[], restaurantId: string | null) {
    this.id = id;
    this.menus = menus;
    this.restaurantId = restaurantId;
  }
}

File: MenuVoting.Domain/Entities/MenuPool.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace MenuVoting.Domain.Entities;
public class MenuPool : IKeyedEntity<Guid>
{
	public Guid Id { get; set; }
	[ForeignKey(nameof(Entities.Restaurant))]
	public Guid RestaurantId { get; set; }
	virtual public Restaurant? Restaurant { get; set; }
	public ICollection<Menu>? Menus { get; set; }
	public DateOnly Date { get; set; }
}
